design/KWOK.md

```diff
@@ -0,0 +1,45 @@
+# Support running on KWOK cluster
+
+## Ticket
+https://runai.atlassian.net/browse/RUN-16744
+
+## Motivation
+The KWOK cluster enables running fake pods on fake nodes, offering an opportunity to reduce the cost of running scale tests on real nodes. 
+This initiative aims to extend support for our system to operate seamlessly within the KWOK cluster environment, so we can reduce the cost of running scale tests on real nodes.
+
+## Assumptions
+- Pods running on fake nodes are considered fake pods. These pods' containers will not execute.
+
+## Requirements
+- Ensure labels and metrics are exposed in the same manner as on real nodes.
+
+## Gaps
+- Status Exporter
+  - The current deployment as a DaemonSet is incompatible with fake nodes.
+- RunAI's Node Exporter
+  - The current deployment as a DaemonSet is incompatible with fake nodes.
+- Device Plugin
+  - The current deployment as a DaemonSet is incompatible with fake nodes. We might want to not supoprt it on fake nodes and require manual node resources update.
+
+## Design
+- [ ] Implement a single monolithic service named `status-exporter` to handle all exportation logic when GPU nodes are fake. This service will be disabled be default, and will be manually enabled when running on kwok cluster. This service will encompass the following:
+  - [ ] Metrics
+    - [ ] Export the same as today, with the following label enrichments (<pod> refers to the dcgm-exporter fake pod):
+      - [ ] `container="nvidia-dcgm-exporter"`
+      - [ ] `instance="<pod-ip>:9400"`
+      - [ ] `job="nvidia-dcgm-exporter"`
+      - [ ] `pod="<pod-name>"`
+      - [ ] `service="nvidia-dcgm-exporter"`
+  - [ ] FileSystem
+    - [ ] Directly export Node Exporter's metrics instead of exporting to the FileSystem, including:
+      - [ ] `runai_pod_gpu_utilization` with labels `pod_uuid` and `gpu`
+      - [ ] `runai_pod_gpu_memory_used_bytes` with labels `pod_uuid` and `gpu`
+  - [ ] Labels
+    - [ ] Ensure consistent label exportation.
+- [ ] Add a ServiceMonitor for the new service, and set `honorLabels: true` on it (so we can fake multiple exporters).
```

Comment:
this section was too technical for me to read and digest without frontal explanation